pints/_mcmc/__init__.py

```diff
@@ -684,14 +684,14 @@ def run(self):
                 # Single chain
 
                 # Check and update the individual chains
-                xs_iterator = iter(xs)
                 fxs_iterator = iter(fxs)
                 for i in list(active):  # new list: active may be modified
-                    x = next(xs_iterator)
-                    fx = next(fxs_iterator)
-                    y = self._samplers[i].tell(fx)
+                    reply = self._samplers[i].tell(next(fxs_iterator))
+
+                    if reply is not None:
+                        # Unpack
```

Comment:
Maybe say something like
`# Unpack (current position, evaluated value, acceptance of last evaluated sample)`